docs/source/model_doc/reformer.rst

```diff
@@ -0,0 +1,114 @@
+Reformer
+----------------------------------------------------
+**DISCLAIMER:** This model is still a work in progress, if you see something strange,
+file a `Github Issue <https://github.com/huggingface/transformers/issues/new?assignees=&labels=&template=bug-report.md&title>`_
+
+Overview
```

Comment:
Added 3 sections explaining a bit more detailed how different tricks employed in the Reformer work.